test/framework/k8s_cluster.go

```diff
@@ -1249,66 +1249,200 @@ func (c *K8sCluster) K8sVersionCompare(otherVersion string, baseMessage string)
 	return version.Compare(semver.MustParse(otherVersion)), fmt.Sprintf("%s with k8s version %s", baseMessage, version)
 }
 
-func printDetailedPodInfo(testingT testing.TestingT, kubectlOptions *k8s.KubectlOptions, podError error, pod v1.Pod) error {
+type appInstallation struct {
+	Name      string
+	Namespace string
+	Replicas  int
+}
+
+func printDeploymentDetails(testingT testing.TestingT, kubectlOptions *k8s.KubectlOptions, namespace string, name string) {
+	deploy, err := k8s.GetDeploymentE(testingT, kubectlOptions, name)
+	if err != nil {
+		// might not be a Deployment, let's ignore it
+		return
+	}
+
+	deployDetails := deploymentDetails{Namespace: namespace,
+		Kind:          "Deployment",
+		objectDetails: newObjectDetails(name),
+		ReplicaSets:   make([]*objectDetails, 0),
+		Pods:          make([]*podDetails, 0),
+	}
+	deployDetails.Conditions = fromDeploymentCondition(deploy.Status.Conditions)
+	deployDetails.Events = getObjectEvents(testingT, kubectlOptions, "Deployment", name)
+
+	replicaSets, _ := k8s.ListReplicaSetsE(testingT, kubectlOptions, metav1.ListOptions{
+		LabelSelector: "app=" + name,
+	})
+	for _, ars := range replicaSets {
+		rsDetails := newObjectDetails(ars.Name)
+		rsDetails.Conditions = fromReplicaSetCondition(ars.Status.Conditions)
+		rsDetails.Events = getObjectEvents(testingT, kubectlOptions, "ReplicaSet", ars.Name)
+		deployDetails.ReplicaSets = append(deployDetails.ReplicaSets, rsDetails)
+	}
+
+	pods, _ := k8s.ListPodsE(testingT, kubectlOptions, metav1.ListOptions{
+		LabelSelector: "app=" + name,
+	})
+	for _, pod := range pods {
+		pDetail := &podDetails{objectDetails: newObjectDetails(pod.Name)}
+		pDetail.Conditions = fromPodCondition(pod.Status.Conditions)
+		pDetail.Events = getObjectEvents(testingT, kubectlOptions, "Pod", pod.Name)
+		deployDetails.Pods = append(deployDetails.Pods, pDetail)
+	}
+
+	deployDetailsJson, err := json.Marshal(deployDetails)
+	if err != nil {
+		deployDetailsJson = []byte(fmt.Sprintf("error marshaling deployment details: '%s'", err.Error()))
+	}
+
+	logger.Default.Logf(testingT, "Detail info of deployment '%s/%s': %s",
+		namespace, name, string(deployDetailsJson),
+	)
+}
+
+func printPodDetails(testingT testing.TestingT, kubectlOptions *k8s.KubectlOptions, podError error, pod v1.Pod) error {
 	if podError == nil {
 		return podError
 	}
 
-	podCond, _ := json.Marshal(pod.Status.Conditions)
-	podLogs, _ := k8s.GetPodLogsE(testingT, kubectlOptions, &pod, "")
-	if podLogs != "" {
-		logLines := strings.Split(podLogs, "\n")
-		if len(logLines) > 30 {
-			podLogs = strings.Join(logLines[len(logLines)-30:], "\n")
-		}
-		podLogs = "; last lines of pod logs: " + podLogs
+	pDetails := &podDetails{objectDetails: newObjectDetails(pod.Name)}
+	pDetails.Conditions = fromPodCondition(pod.Status.Conditions)
```

Comment:
Can we declare these all at once above when we initialize `pDetails`?

Reply:
Sure, just updated code.